lambda_code/src/fhir_service.py

```diff
@@ -78,18 +92,25 @@ def delete_immunization(self, imms_id) -> Immunization:
         imms = self.immunization_repo.delete_immunization(imms_id)
         return Immunization.parse_obj(imms)
 
-    def search_immunizations(self, nhs_number: str, disease_type: str) -> FhirList:
+    def search_immunizations(self, nhs_number: str, disease_type: str) -> FhirBundle:
         """find all instances of Immunization(s) for a patient and specified disease type.
         Returns List[Immunization]
         """
         # TODO: is disease type a mandatory field? (I assumed it is)
         #  i.e. Should we provide a search option for getting Patient's entire imms history?
         resources = self.immunization_repo.find_immunizations(nhs_number, disease_type)
-
         patient = self.pds_service.get_patient_details(nhs_number)
-
-        entries = [Immunization.parse_obj(handle_s_flag(imms, patient)) for imms in resources]
-        return FhirList.construct(entry=entries)
+        entries = [
+            BundleEntry(resource=Immunization.parse_obj(handle_s_flag(imms, patient))) for imms in resources
+        ]
+        fhir_bundle = FhirBundle(
+            resourceType="Bundle",
+            type="searchset",  # Set the type to "searchset"
+            entry=entries,
+        )
+        url = f"{get_service_url()}/Immunization?patient={nhs_number}&diseaseType={disease_type}"
```

Comment:
Shouldn't `patient` be `nhsNumber`?
Or better yet, could the URL be taken from the event so we know for sure it's exactly what was used?

Reply:
Self link in Searchset Bundle can only be a get, thats why i prepared it such a way that irrespective of POST/GET call, it will prepare a self link which will be get. Yeah you are right, that parameter should be nhsNumber